_posts/2017-06-13-how-do-I-convince.md

```diff
@@ -0,0 +1,28 @@
 feedback loop when it comes to the correctness of our code? And how do we make it repeatable? Hmm... Maybe a set of automated tests could be very helpful. In minutes, if not seconds, we can know if our code works. Yes, that would be cool because today it takes us at least 2 weeks for our QA team to test the system. We also need to have “code freezes” while they do their testing. Yeah, that’s quite wasteful when we compare the size of feedback loop and how repeatable the process is. OK, I’m convinced. Now, how do we build this test suite? How do we make sure we don’t over engineer and only build enough to satisfy business requirements? How do we get feedback in our design? Hmm... Maybe Test-Driven Development could be a very good practice. True. This way we can move along being confident that we haven’t missed anything and everything we’ve done is tested. Hey, we don’t ever need to bother with code coverage, as we would be always if not at 100%, at least very close to it. Yes, but it would be nice to share knowledge about different parts of the code base with all members of the team, and also make sure we are all happy with the quality of code we are producing. Hmm... Maybe we should start pair programming and rotate pairs often. Yes, and we could have another pair reviewing our code once we are done. OK. So we can use develop our code use different branches and merge the code once it is reviewed, assuming that no issues were found.  Yes, and to make reviews faster and easier (we don’t want to waste anyone’s time), we could work on very short-lived branches and make sure each branch focuses on a very specific and single behaviour. Cool, sounds a good idea. And how can we know that the work being done in parallel by different people is not breaking anything else? Hmm... Maybe we could make it easy to continuously integrate our code and run all of our tests every time it happens. Yes, that would give us an immediate feedback on our change. Even better if we had this check when we raise a pull request, at branch level and not after the code is merged. If the build fails, we know we can’t merge our code and this way we can always keep the main branch clean and ready to be deployed. Yes, that makes sense. I love that. Right, so if all production code has been written to satisfy an existing test, it has been written and reviewed by different pairs, it was successfully integrated and tested, then we could potentially go live, right? Sure, why not? OK, maybe we could automate our deployment pipeline, removing the need for manual steps and making the process repeatable. Yes, this way we could trigger it as many times as we want. Even if we ever need to make an emergency fix in production (shit sometimes happen you know) it would be easy and safe to do it as everything is automated. From our experience, when we quickly need to fix a bug, we may create other bugs. With all our practices in places, this would be a very unlikely scenario. Sounds like a plan. 
```

Comment:
"So we can use develop our code use different branches and merge the code once it is reviewed" - there's a typo in there but I can't figure how to fix it :)